components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/RoleBasedInternalScopeValidator.java

```diff
@@ -173,9 +218,16 @@ private List<String> getRolesOfTheUser(AuthenticatedUser authenticatedUser) thro
                     = (AbstractUserStoreManager) realmService.getTenantUserRealm(tenantId).getUserStoreManager();
 
             String userName = userStoreManager.getUserNameFromUserID(authenticatedUser.getUserId());
+            // If the username is empty try to set it as authenticated username.
+            if (StringUtils.isEmpty(userName)) {
```

Comment:
which cases can this be empty?

Reply:
if we take this feature (FIdp role-based authentication), we manipulated the user tenant which results in the user id not being set for the authenticated user.